draft-js-side-toolbar-plugin/src/index.d.ts

```diff
@@ -0,0 +1,47 @@
+
+import { EditorPlugin } from "draft-js-plugins-editor";
+import { ComponentType, ReactNode } from "react";
+import { EditorState } from "draft-js";
+
+export interface SideToolbarPluginTheme {
+    buttonStyles?: {
+        buttonWrapper?: string;
+        button?: string;
+        active?: string;
+        separator?: string;
+    };
+    blockTypeSelectStyles?: {
+        blockType?: string;
+        spacer?: string;
+        popup?: string;
+    };
+    toolbarStyles?: {
+        wrapper?: string;
+    };
+}
+
+type SideToolbarPosition = "left" | "right";
+
+export interface SideToolbarPluginConfig {
+    theme?: SideToolbarPluginTheme;
+    position?: SideToolbarPosition;
+}
+
+export interface SideToolbarChildrenProps {
+    theme: SideToolbarPluginTheme["buttonStyles"];
+    getEditorState: () => EditorState;
+    setEditorState: (editorState: EditorState) => void;
+    onOverrideContent: (content: ComponentType<SideToolbarChildrenProps>) => void;
+}
+
+export interface SideToolbarProps {
+    children?: ComponentType<SideToolbarChildrenProps>;
```

Comment:
This should be a function, not a component.

Reply:
it's the same, a function who receive props and return a ReactNode is the same that if I defined it as a ComponentType who receive props typed by SideToolbarChildrenProps

Reply:
No, it's not.
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L71

This function is just called. Not instantiated with React.createElement. If you use ComponentType user can pass class components. Classes cannot be called without `new` keyword.
https://github.com/draft-js-plugins/draft-js-plugins/blob/040436205cdf59236267c4dca40c80d41b8b8f9c/draft-js-side-toolbar-plugin/src/components/BlockTypeSelect/index.js#L62

Reply:
Totally right, fixed. Thank you so much for the information and the effort :)